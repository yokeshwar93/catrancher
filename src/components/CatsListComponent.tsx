import React from "react";
import {makeStyles} from "@material-ui/core";
import Grid from '@material-ui/core/Grid';
import Paper from '@material-ui/core/Paper';
import {CatsList, CatsListProps} from "../models/modelTypes";

const useStyles = makeStyles((theme) => ({
    root: {
        flexGrow: 1,
        marginTop: '1%',
        marginLeft: '2%'
    },
    paper: {
        height: '95%',
        width: '80%',
        cursor: 'pointer',
        borderRadius: '15px',

    },
    selected: {
        background: 'lightgray'
    },
    control: {
        padding: theme.spacing(2),
    },
    gridContainerStyle: {
        height: '25vh',
        width: '10vw'
    },
    imageStyle: {
        height: '95%',
        width: '95%',
        marginLeft: 'auto',
        marginRight: 'auto',
        display: 'block'
    }
}));



const CatsListComponent = (props : CatsListProps) => {
    const classes = useStyles();
    const { catsList, selectedCats, handleOnCatSelect } = props;

    const isCatSelected = (catId : string) : boolean => {
        return selectedCats.findIndex(data => data.id === catId) !== -1
    }
    return (
        <Grid container className={classes.root} spacing={2}>
            <Grid item xs={12}>
                {[0, 4, 8].map(value => (
                    <Grid container key={`${value}`} justifyContent="center" spacing={3}>
                        {catsList.slice(value, value +4).map((cat : CatsList) => (
                            <Grid className={classes.gridContainerStyle} key={cat.id} item>
                                <Paper onClick={() => handleOnCatSelect(cat.id, cat.imageUrl)} className={`${classes.paper} ${isCatSelected(cat.id) ? classes.selected : ''}`} elevation={0} >
                                    <img className={classes.imageStyle} src={cat.imageUrl} alt={`Cat image_${cat.id}`} />
                                </Paper>
                            </Grid>
                        ))}
                    </Grid>
                ))
                }
            </Grid>
        </Grid>
    )
}

export default CatsListComponent;
