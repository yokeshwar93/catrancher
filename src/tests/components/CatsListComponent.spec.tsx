import React from 'react';
import { render } from '@testing-library/react';
import { mockStore } from "../store/reducer/mockData";
import CatsListComponent from "../../components/CatsListComponent";

describe('renders Catslist component', () => {

    it('render first grid with cats image', () => {
        const { container } = render(
                <CatsListComponent catsList={mockStore.catsList} handleOnCatSelect={jest.fn()} selectedCats={mockStore.selectedCats} />
        );
        expect(container).toMatchSnapshot();

    })

});
