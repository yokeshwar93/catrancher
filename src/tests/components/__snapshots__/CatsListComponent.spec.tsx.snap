// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Catslist component render first grid with cats image 1`] = `
<div>
  <div
    class="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-justify-content-xs-center"
      >
        <div
          class="MuiGrid-root makeStyles-gridContainerStyle-5 MuiGrid-item"
        >
          <div
            class="MuiPaper-root makeStyles-paper-2 makeStyles-selected-3 MuiPaper-elevation0 MuiPaper-rounded"
          >
            <img
              alt="Cat image_1ttg"
              class="makeStyles-imageStyle-6"
              src="Image url"
            />
          </div>
        </div>
        <div
          class="MuiGrid-root makeStyles-gridContainerStyle-5 MuiGrid-item"
        >
          <div
            class="MuiPaper-root makeStyles-paper-2 makeStyles-selected-3 MuiPaper-elevation0 MuiPaper-rounded"
          >
            <img
              alt="Cat image_2ttg"
              class="makeStyles-imageStyle-6"
              src="Image url"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-justify-content-xs-center"
      />
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-justify-content-xs-center"
      />
    </div>
  </div>
</div>
`;

exports[`renders loader by default render loader with intitial store 1`] = `
<div>
  <div
    class="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-justify-content-xs-center"
      >
        <div
          class="MuiGrid-root makeStyles-gridContainerStyle-5 MuiGrid-item"
        >
          <div
            class="MuiPaper-root makeStyles-paper-2 makeStyles-selected-3 MuiPaper-elevation0 MuiPaper-rounded"
          >
            <img
              alt="Cat image_1ttg"
              class="makeStyles-imageStyle-6"
              src="Image url"
            />
          </div>
        </div>
        <div
          class="MuiGrid-root makeStyles-gridContainerStyle-5 MuiGrid-item"
        >
          <div
            class="MuiPaper-root makeStyles-paper-2 makeStyles-selected-3 MuiPaper-elevation0 MuiPaper-rounded"
          >
            <img
              alt="Cat image_2ttg"
              class="makeStyles-imageStyle-6"
              src="Image url"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-justify-content-xs-center"
      />
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-justify-content-xs-center"
      />
    </div>
  </div>
</div>
`;
